flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/nodes/FlinkRelNode.scala

```diff
@@ -32,6 +32,18 @@ import scala.collection.JavaConversions._
   */
 trait FlinkRelNode extends RelNode {
 
+  /**
+    * Return true if this rel is a deterministic RelNode which is guaranteed to
+    * always output the same result given the same input.
+    *
+    * <p>NOTES: A deterministic rel should not contain non-deterministic `SqlOperator`
+    * and dynamic function `SqlOperator`.
+    * e.g.
+    * `Filter` is non-deterministic if its condition contains non-deterministic udf,
+    * `Aggregate` is non-deterministic if its aggCalls contain non-deterministic `SqlAggFunction`.
+    */
+  def isDeterministic: Boolean
```

Comment:
I'm not sure we should add this interface to this base class.

Reply:
I have deleted this method. 
the original thought about this method is we do not know whether the reused plan with non-deterministic operator is the plan expected by the user, so we add an option to let user to decide it.